src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs

```diff
@@ -972,6 +975,111 @@ internal string GetPropertyRawText()
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if trying to find a text match on a JSON token that is not a string
+        /// (i.e. other than <see cref="JsonTokenType.String"/> or <see cref="JsonTokenType.PropertyName"/>).
+        /// <seealso cref="TokenType" />
+        /// </exception>
+        /// <remarks>
+        /// If the look up text is invalid UTF-8 text, the method will return false since you cannot have 
+        /// invalid UTF-8 within the JSON payload.
+        /// </remarks>
+        /// <remarks>
+        /// The comparison of the JSON token value in the source and the look up text is done by first unescaping the JSON value in source,
+        /// if required. The look up text is matched as is, without any modifications to it.
+        /// </remarks>
+        public bool ValueEquals(ReadOnlySpan<byte> utf8Text)
+        {
+            CheckValidInstance();
+
+            if (TokenType == JsonTokenType.Null)
+            {
+                return false;
+            }
+
+            return TextEqualsHelper(utf8Text);
+        }
+
+        /// <summary>
+        /// Compares the string text to the unescaped JSON token value in the source and returns true if they match.
+        /// </summary>
+        /// <param name="text">The text to compare against.</param>
+        /// <returns>True if the JSON token value in the source matches the look up text.</returns>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if trying to find a text match on a JSON token that is not a string
+        /// (i.e. other than <see cref="JsonTokenType.String"/> or <see cref="JsonTokenType.PropertyName"/>).
+        /// <seealso cref="TokenType" />
+        /// </exception>
+        /// <remarks>
+        /// If the look up text is invalid or incomplete UTF-16 text (i.e. unpaired surrogates), the method will return false
+        /// since you cannot have invalid UTF-16 within the JSON payload.
+        /// </remarks>
+        /// <remarks>
+        /// The comparison of the JSON token value in the source and the look up text is done by first unescaping the JSON value in source,
+        /// if required. The look up text is matched as is, without any modifications to it.
+        /// </remarks>
+        public bool ValueEquals(ReadOnlySpan<char> text)
+        {
+            CheckValidInstance();
+
+            if (TokenType == JsonTokenType.Null)
+            {
+                // This is different than Length == 0, in that it tests true for null, but false for ""
+                return text == default;
+            }
+
+            return TextEqualsHelper(text);
+        }
+
+        internal bool TextEqualsHelper(ReadOnlySpan<byte> utf8Text, bool isPropertyName = false)
+        {
+            // CheckValidInstance(); // TODO: Check if needed. called from JsonProperty.NameEquals
```

Comment:
The test would be

```C#
JsonProperty prop = default;
Assert.Throws<InvalidOperationException>(() => prop.NameEquals("hello"));
```

Without this it'd be a NullReferenceException.

Since TokenType won't throw, and Null couldn't happen if CheckValidInstance would throw, uncommenting this one (to make the property behave appropriately) and replacing the calls in the public method with `// CheckValidInstance is done in the helper` seems reasonable.

Reply:
That's awesome thanks for the feedback.